core/src/main/java/org/apache/calcite/rel/rules/JoinDeriveIsNotNullFilterRule.java

```diff
@@ -63,9 +63,11 @@ public JoinDeriveIsNotNullFilterRule(Config config) {
 
     final ImmutableBitSet.Builder notNullableKeys = ImmutableBitSet.builder();
     RelOptUtil.conjunctions(join.getCondition()).forEach(node -> {
```

Comment:
Is there any benefit in splitting the condition in conjunctions and testing strongness individually for each? Maybe we can remove this decomposition.